src/main/java/smokefree/domain/Initiative.java

```diff
@@ -206,5 +216,46 @@ private void assertCurrentUserIsManager(MetaData metaData) {
                 "You are not a manager of this playground"));
     }
 
+    private void validateMaximumPlaygroundCapacity() {
+        if(initiativeProjection.playgrounds().size() >= initiativeProjection.MAXIMUM_PLAYGROUNDS_ALLOWED) {
+            throw new DomainException("MAXIMUM_PLAYGROUNDS_CAPACITY_REACHED",
+                    "Can not add more than " + initiativeProjection.MAXIMUM_PLAYGROUNDS_ALLOWED + " playgrounds",
+                    "Sorry, Maximum playgrounds capacity is reached, please contact helpline");
+        }
+    }
+
+    private void validateDuplicatePlaygroundNames(String playgroundName) {
+        initiativeProjection.playgrounds().entrySet()
+                .stream()
+                .filter( playgroundEntry -> playgroundEntry.getValue().getName().equals(playgroundName))
+                .map(playgroundEntry -> playgroundEntry.getValue())
+                .findFirst()
+                .ifPresent( p -> {
+                    throw new DomainException("DUPLICATE_PLAYGROUND_NAME",
+                            "Playground " + playgroundName + " does already exist, please choose a different name",
```

Comment:
I am getting the feeling you are mixing up the order of the parameters in the DomainException. The second parameter is the technical message that is not sent to the front-end and the third is a message that is sent to the front-end. But your second message is more user-friendly than the third. You perhaps you intended them to be the other way around?

Reply:
DomainExceptions are not directly propagated from AXON to GraphQL DataFetchers. it's translated to ExecutionException That's because these DomainExceptions and ConstraintViolationException are caught in ConfigurableDataFetcherExceptionHandler.java.

Reply:
Ok, but how are these exceptions presented to the frontend?

Reply:
the exceptions presented as errors,
```result.put("errors", executionResult
                .getErrors()
                .stream()
                .map(error -> determineErrorAttributes(error))
                .collect(toList()));